examples/js/loaders/BasisTextureLoader.js

```diff
@@ -109,7 +112,36 @@ THREE.BasisTextureLoader.prototype = Object.assign( Object.create( THREE.Loader.
 
 		loader.load( url, ( buffer ) => {
 
-			this._createTexture( buffer )
+			var taskKey = JSON.stringify( url );
+
+			// Check for an existing task using this buffer. A transferred buffer cannot be transferred
+			// again from this thread.
+			if ( THREE.BasisTextureLoader.taskCache.has( buffer ) ) {
+
+				var cachedTask = THREE.BasisTextureLoader.taskCache.get( buffer );
+
+				if ( cachedTask.key === taskKey ) {
+
+					return cachedTask.promise.then( onLoad ).catch( onError );
+
+				} else if ( buffer.byteLength === 0 ) {
+
+					// Technically, it would be possible to wait for the previous task to complete,
+					// transfer the buffer back, and decode again with the second configuration. That
+					// is complex, and I don't know of any reason to decode a Basis buffer twice in
+					// different ways, so this is left unimplemented.
+					throw new Error(
+
+						'THREE.BasisTextureLoader: Unable to re-decode a buffer with different ' +
+						'settings. Buffer has already been transferred.'
+
+					);
+
+				}
```

Comment:
I think you can leave out the `taskKey`, and just assume that any task for the same buffer is equivalent. No need to throw an error. Unlike Draco geometry, decoding a Basis texture currently depends only on hardware capabilities and not other input parameters (for now at least).

Reply:
@donmccurdy used url as a task key.

Reply:
Thanks! I'd just remove the `if ( cachedTask.url === url ) {` check entirely, if the buffer matches that's enough. Otherwise we're basically comparing two pairs of interchangeable keys.